preupg/application.py

```diff
@@ -732,19 +737,17 @@ def run(self):
                     ' '.join(SystemIdentification.get_convertors())))
                 return ReturnValues.MISSING_TEXT_CONVERTOR
 
-        if not self.conf.scan and not self.conf.contents and \
-                not self.conf.list_rules:
-            self.conf.scan = self.get_default_module_set_dirname()
-            if not self.conf.scan:
-                return ReturnValues.SCENARIO
+        if self.conf.scan and self.conf.contents:
```

Comment:
This should be handled inside `preupg/cli.py`.

Reply:
Moved to `preupg/cli.py/resolve_option_dependencies()`.